src/software/ai/hl/stp/skill/pass/pass_skill_fsm.cpp

```diff
@@ -23,21 +32,23 @@ bool PassSkillFSM::passFound(const Update& event)
                                    best_pass_so_far_->pass.estimatePassDuration();
 }
 
-bool PassSkillFSM::passReceived(const Update& event)
+bool PassSkillFSM::passReceived(const SuspendedUpdate& event)
 {
-    auto friendly_robots =
-        event.common.world_ptr->friendlyTeam().getAllRobotsExcept({event.common.robot});
-
-    return std::any_of(
-        friendly_robots.begin(), friendly_robots.end(), [&](const Robot& robot) {
-            return robot.isNearDribbler(event.common.world_ptr->ball().position());
-        });
+    const auto ai_config       = event.strategy->getAiConfig();
+    const auto friendly_robots = event.world_ptr->friendlyTeam().getAllRobots();
+
+    return event.world_ptr->ball().velocity().length() <
+               ai_config.ai_parameter_config().ball_is_kicked_m_per_s_threshold() &&
+           std::any_of(
+               friendly_robots.begin(), friendly_robots.end(),
+               [&](const Robot& robot)
+               { return robot.isNearDribbler(event.world_ptr->ball().position()); });
```

Comment:
consider abstracting this into a helper function that we can use in `ReceiverFSM`

Reply:
Tried doing this and it didn't make the code much cleaner. We have to set up and pass in a lot of parameters, so the helper function is essentially just a wrapper around std::any_of